### DISCUSSION TITLE
Help!! ERR_REQUIRE_ESM in vscode-remark but works on remark-cli

### DISCUSSION THREAD

**Discussion by davorpa** [remarkjs/remark#846] [Q&A] [answered]
Hello community

I don't know if it's really a bug or if I have something wrong configured at my environment, so I prefer to ask first before scale it.

I am trying to migrate from `markdown-all-in-one`/`markdown-lint` to `remark` because I love the powerful flexibility gives plugins. In this progress I can't get the VSCode plugin works, no matter how hard I try. Solving previous errors, I'm stucked at this point where the output throws:

```log
[Remark]
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: $WORKSPACE_PATH\node_modules\remark-lint\index.js
require() of ES modules is not supported.
require() of $WORKSPACE_PATH\node_modules\remark-lint\index.js from $HOME\.vscode\extensions\unifiedjs.vscode-remark-1.3.0\out\extension.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from $WORKSPACE_PATH\node_modules\remark-lint\package.json.
```

## My enviroment

> Version: 1.60.0 (system setup)
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:41:52.311Z
Electron: 13.1.8
Chrome: 91.0.4472.164
**Node.js: 14.16.0**
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 6.1.7601 (W7 Ultimate)
**unifiedjs.vscode-remark: 1.3.0**

At console I have installed Node 12.x (due to OS restrictions with 14.x) and `remark-cli` works like a charm colaborating with `husky`/`lint-staged`

```shell
$ node --version
v12.22.5
```

## My config files

As it could check, I already have the addon updated to latest version `1.3.0`. It was the way to solve earlier errors.

I have all configured using a **`.remarkrc`** in the project's root folder

```json
{
    "plugins": [
        ["lint"],
        ["lint-alphabetize-lists"],
        ["lint-final-newline", true],
        ["lint-heading-style", "atx"],
        ["lint-no-duplicate-headings", true],
        ["lint-no-emphasis-as-heading", true],
        ["lint-no-heading-content-indent", true],
        ["lint-no-heading-indent", true],
        ["lint-no-heading-punctuation", true],
        ["lint-no-literal-urls", true],
        ["lint-no-tabs", true],
        ["lint-strong-marker", "consistent"],
        ["lint-unordered-list-marker-style", "*"],
        ["lint-blank-lines-1-0-2"],
        ["lint-books-links"],
        ["lint-no-empty-sections"],
        ["lint-no-empty-url"],
        ["lint-no-url-trailing-slash"]
    ]
}
```

And of course installed each package as seen in **`package.json`**

```json
{
    "scripts": {
        "lint": "remark ."
    },
    "devDependencies": {
        "remark-cli": "^10.0.0",
        "remark-lint": "^9.0.1",
        "remark-lint-alphabetize-lists": "^3.0.0",
        "remark-lint-blank-lines-1-0-2": "^3.0.0",
        "remark-lint-books-links": "^2.1.0",
        "remark-lint-final-newline": "^2.0.1",
        "remark-lint-heading-style": "^3.0.1",
        "remark-lint-no-duplicate-headings": "^3.0.1",
        "remark-lint-no-emphasis-as-heading": "^3.0.1",
        "remark-lint-no-empty-sections": "^4.0.0",
        "remark-lint-no-empty-url": "^3.0.1",
        "remark-lint-no-heading-content-indent": "^4.0.1",
        "remark-lint-no-heading-indent": "^4.0.1",
        "remark-lint-no-heading-punctuation": "^3.0.1",
        "remark-lint-no-literal-urls": "^3.0.1",
        "remark-lint-no-tabs": "^3.0.1",
        "remark-lint-no-url-trailing-slash": "^3.0.1",
        "remark-lint-strong-marker": "^3.0.1",
        "remark-lint-unordered-list-marker-style": "^3.0.1"
    }
}
```

Is it possible that it is some incompatibility of versions? But which one? I don't know :worried: 

Thanks in advance :bow:

**Comment by wooorm** [comment, marked as answer]
Unfortunately VSCode/Electron don’t support ESM yet, whereas all remark plugins have migrated to it.
The only solution for now is to use the previous version of all packages, or only use remark-cli!

**Reply by davorpa** [reply]
Thanks!! Now I'm a bit more calm. So the problem is Node 14.x

CLI combined with husky/lint-stage is fine for me, or even more aggresive enabling watch and output options to autofix on saving files.

I'll see to downgrade versions one by one too. What version is a good start?

**Reply by wooorm** [reply]
- All the last versions of all packages maintained by unified and remark for downgrading
- The problem isn’t Node, Node 12, 14, and 16 all support ESM. The problem is Electron

**Reply by davorpa** [reply]
Thanks for all your answers. Now I understand better :rocket:

Have a good day :hugs:

**Reply by davorpa** [reply]
Talking about ESM... if this is helpfull for others, are discussing about it
- [x] standard-things/esm#868
